### PR TITLE
Update MALI version and Greenland mesh

### PR DESCRIPTION
**Pull request by matthewhoffman**
This merge improves support for a dynamic Greenland ice sheet in E3SM using MALI.  This includes: 
* Updating the version of Albany used on Cori to allow velocity solver improvements
* Update to the GIS 1km initial condition file
* Removing unneeded land model warnings

 These changes have been tested in both IG and BG compsets and run for one year at 5 day timesteps
 and 5 day glc coupling intervals. Results were examined qualitatively and deemed to be reasonable
 and realistic.

Other features not generally used in E3SM:
* Change from using beta as basal friction field to muFriction
* Improvements to calving and facemelting parameterizations that allow for exploration of more sophisticated calving treatment (off in E3SM)
* Separates a flood fill routine for general use (BFB under all configurations)
* A new namelist option to restore thickness after advection needed for some idealized configurations (not used in E3SM)
* Improvements and fixes to the subglacial hydrology model (not currently used in E3SM)

Bug fixes:
 * effective viscosity calculation caused a divide by zero in some situations and is now fixed
* Fix bug to ensure time-varying forcing is read on initial time
* Missing error variables are fixed that led to improper error handling in some cases
* Damage was not being advected - that is now fixed (damage evolution is not currently used in E3SM)

[NML]
[non-BFB] for configurations with MALI

### CODE REVIEW

**Comment by jonbob** [comment]
Thanks @matthewhoffman -- and congrats for getting this pulled together. I'm happy to run the automated scripts and commit the corresponding changes to this branch, if that helps as a starting point

**Comment by matthewhoffman** [comment]
@jonbob , thanks, that would be great.  Let me know if anything comes up or if I can contribute to that.

**Comment by jonbob** [comment]
@matthewhoffman - I looked at the changes and the only thing I see that I need to be aware of when I run the scripts is this change to defaults:
```
-		<dim name="maxTracersAdvect" definition="1" units="unitless"
+ 		<dim name="maxTracersAdvect" definition="2" units="unitless"
```
??

**Comment by matthewhoffman** [comment]
@jonbob , that change is needed, but there also should be a handful of new namelist options.  I can identify a list if that would help.

**Comment by jonbob** [comment]
New ones aren't trouble -- just changes to default settings require some special attention

**Comment by jonbob** [comment]
@matthewhoffman - we may need a more descriptive name for the branch. Merging from develop sounds a bit shaky or unclear

**Comment by jonbob** [comment]
@matthewhoffman - I ran the automated scripts but I don't think I have permissions to push the updates back to MALI-Dev

**Comment by matthewhoffman** [comment]
@jonbob , I've added you as a member to the MALI-Dev org, so you should be able to push to branches there now.  I think our `develop` branch is protected, though, so you might not be able to push directly to it.  Let me know if that is the case, and I can either add you commit as a new branch or I can relax that restriction.

Regarding the name of the branch `develop`, I asked @rljacob about that last year, and at the time he was fine with us using that name (given that it is in another fork).  @rljacob , do you still think it's ok that we are using a branch name called `develop` inside the `MALI-Dev` organization?  It looks like this in a PR back to E3SM:
`matthewhoffman wants to merge 70 commits into E3SM-Project:master from MALI-Dev:develop `

**Comment by jonbob** [comment]
Thanks @matthewhoffman - I was working on anvil, so once it's back up I'll try again

**Comment by rljacob** [comment]
I'd suggest making a branch off of develop, something like mhoffman/mali/develop-YYMMDD and then merge that to E3SM.

File: components/elm/src/biogeophys/BalanceCheckMod.F90 [hunk 1]
```diff
@@ -344,7 +344,7 @@ subroutine ColWaterBalanceCheck( bounds, num_do_smb_c, filter_do_smb_c, &
        do fc = 1,num_do_smb_c
           c = filter_do_smb_c(fc)
           g = col_pp%gridcell(c)
-          write(iulog,*)'WARNING:  glc_dyn_runoff_routing = ', glc_dyn_runoff_routing(g)  ! TKT
+!          write(iulog,*)'WARNING:  glc_dyn_runoff_routing = ', glc_dyn_runoff_routing(g)  ! TKT
```
Comment (rljacob) [inline components/elm/src/biogeophys/BalanceCheckMod.F90:347]: should this be part of this merge?

Comment (matthewhoffman) [reply components/elm/src/biogeophys/BalanceCheckMod.F90:347]: Yes, @stephenprice checked with @tktesfa2014 about this, and the determination is it was a debugging write that had been meant to be commented but was still active.  It was leading to thousands of write statements when running compsets with active MALI.  Normally we make a rule to not touch anything outside of the MALI directory in our fork, but given the simple nature of this change we included it in Steve's updates (https://github.com/MALI-Dev/E3SM/pull/16).

**Review by jonbob — APPROVED** [review]
Approved based on developer testing and a few I ran on my own

**Comment by matthewhoffman** [comment]
@rljacob , let me know if you need any other info for your review.

**Review by rljacob — APPROVED** [review]

**Comment by matthewhoffman** [comment]
Thanks, Rob!

File: cime_config/machines/cmake_macros/intel_cori-haswell.cmake [hunk 2]
```diff
@@ -1,5 +1,3 @@
-set(ALBANY_PATH "/global/cfs/cdirs/e3sm/software/albany-trilinos/albany-install-2020-08-07")
-string(APPEND FFLAGS " -convert big_endian -assume byterecl -ftz -traceback -assume realloc_lhs -fp-model consistent -fimf-use-svml")
```
Comment (jonbob) [inline cime_config/machines/cmake_macros/intel_cori-haswell.cmake:2]: @matthewhoffman - did you intend to remove the FFLAGS line here? I understand getting rid of the ALBANY_PATH but want to make sure about the rest of it before merging

Comment (matthewhoffman) [reply cime_config/machines/cmake_macros/intel_cori-haswell.cmake:2]: @jonbob , I have pushed an update that fixes this inadvertent deletion (as discussed at https://github.com/MALI-Dev/E3SM/pull/16#discussion_r788184511).  Thanks for catching that!

**Comment by jonbob** [comment]
Testing of trial merge showed expected results for developer tests with MALI on chrysalis:
* SMS.T62_oQU120_ais20.MPAS_LISIO_TEST.chrysalis_intel
* ERS.f09_g16_g.MALISIA.chrysalis_intel

Merged to next

**Comment by jonbob** [comment]
merged to master and expected DIFFs and NML DIFFs blessed

**Pull request merged**